src/tests/test_raft_repl_dev_dynamic.cpp

```diff
@@ -110,28 +116,176 @@ TEST_F(ReplDevDynamicTest, TwoMemberDown) {
         LOGINFO("Member in got all commits");
     }
 
-    if (g_helper->replica_num() == 0 || g_helper->replica_num() == member_in) {
+    if (is_replica_num_in({0, member_in})) {
         // Validate data on leader replica 0 and replica 3
         LOGINFO("Validate all data written so far by reading them replica={}", g_helper->replica_num());
         this->validate_data();
     }
 
-    g_helper->sync_for_cleanup_start(num_members);
-
     if (g_helper->replica_num() == 1) {
         LOGINFO("Start replica 1");
+        db->set_zombie();
         this->start_replica(1);
     }
     if (g_helper->replica_num() == 2) {
         LOGINFO("Start replica 2");
+        db->set_zombie();
+        this->start_replica(2);
+    }
+
+    g_helper->sync_for_cleanup_start(num_members);
+    LOGINFO("TwoMemberDown test done replica={}", g_helper->replica_num());
+}
+
+TEST_F(ReplDevDynamicTest, OneMemberDown) {
+    // replica0(leader) and replica1 up, replica2 is down. Replace replica2 with replica3.
+    // replica0 should be able to baseline resync to replica4(new member).
+    // Write some IO's, replace a member, validate all members data except which is out.
+    LOGINFO("OneMemberDown test started replica={}", g_helper->replica_num());
+    auto db = dbs_.back();
+    auto num_replicas = SISL_OPTIONS["replicas"].as< uint32_t >();
+    auto num_members = SISL_OPTIONS["replicas"].as< uint32_t >() + SISL_OPTIONS["spare_replicas"].as< uint32_t >();
+    uint64_t num_io_entries = SISL_OPTIONS["num_io"].as< uint64_t >();
+
+    // Replace the last member in the group with index(num_replicas - 1) with a spare
+    // replica with index (num_replica). Member id's are 0,...,num_replicas-1, num_replicas,...,N
+    uint32_t member_out = num_replicas - 1;
+    uint32_t member_in = num_replicas;
+
+    g_helper->sync_for_test_start(num_members);
+
+    this->shutdown_replica(2);
+    LOGINFO("Shutdown replica 2");
+
+    std::this_thread::sleep_for(std::chrono::seconds(3));
+    if (g_helper->replica_num() == 0) {
+        // With existing raft repl dev group, write IO's, validate and call replace_member on leader.
+        LOGINFO("Writing on leader num_io={} replica={}", num_io_entries, g_helper->replica_num());
+        this->write_on_leader(num_io_entries, true /* wait_for_commit */);
```

Comment:
I think the `write_on_leader` needs to be execute on r0-r2.
In current implementation  R1/R2 skips the if-clause from L161-L171 and wait on L173.  If R1 is laggy (only commit on R0 and R3),   it is possible the R1 failed during verify

Reply:
agree. in write_on_leader , the real leader will schedule real io, others will wait_for_commit.

Reply:
write_on_leader is nop on other replica's. replica 0 is the leader and we are calling replace member with run_on_leader. I havent observed any replica's lagging in any of this or the original raft repl tests.

Reply:
>write_on_leader is nop on other replica's. 
It is not that "nop" but waiting for IOs to finish before proceeding.

I think all of the UTs except this one follows the pattern.  For example this:
https://github.com/eBay/HomeStore/pull/574/files#diff-8d61e3935e3a521ab097bbc27963ca7e583d90d6a597d1301a0378fa2cf33854R213

Reply:
The reason is we are shutting down replica 2 and it doesnt db instances and write on leader will crash at line auto leader_uuid = dbs_[0]->repl_dev()->get_leader_id();

Reply:
yeah that makes sense, 
```if (g_helper->replica_num() != 2) ```
if this is better?

Reply:
>leader will crash at line auto leader_uuid = dbs_[0]->repl_dev()->get_leader_id();

can we add a check like this `if (is_replica_num_in({0, 1, member_in}))` in write_on_leader

Reply:
write_on_leader is generic and this case is specific for stop .